completion.fish

```diff
@@ -0,0 +1,10 @@
+# fish completion for git fixup
+
+function __fish_git_fixup_target
+    git fixup --no-commit 2>/dev/null | string replace -r '^([0-9a-f]{10})[0-9a-f]* (.*)' '$1\t$2'
```

Comment:
so this only picks the first 10 characters of the commit hash which sure it's probably unique but any particular reason for doing so?

Reply:
I copied this part over from the git completion in fish itself.

https://github.com/fish-shell/fish-shell/blob/master/share/completions/git.fish#L43

The reason given in the comment above this code:

> Hashes we just truncate ourselves to 10 characters, without disambiguating.
> That technically means that sometimes we don't give usable SHAs, but according to https://stackoverflow.com/a/37403152/3150338, that happens for 3 commits out of 600k.
> For fish, at the time of writing, out of 12200 commits, 7 commits need 8 characters.
> And since this takes about 1/3rd of the time that disambiguating takes...

If you'd like I can change it to match zsh's behaviour…

Reply:
no, ok. makes sense to align with the fish conventions for that